paper-linear-progress.html

```diff
@@ -0,0 +1,916 @@
+<!--
+@license
+Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../paper-styles/color.html">
+<link rel="import" href="../neon-animation/neon-animation-runner-behavior.html">
+<link rel="import" href="../neon-animation/animations/transform-animation.html">
+
+<!--
+Material design: [Progress & activity](https://www.google.com/design/spec/components/progress-activity.html)
+
+Linear Progress elements are used to represent an operation that takes a known or unknown (aka
+"indeterminate") amount of time. They can be reversed in order to support RTL layouts and further
+differentiate between different phases of an operation. They can also display a "buffering" state,
+which is useful for operations involving the download of large content (e.g. media).
+
+Example:
+
+    <paper-linear-progress value="10"></paper-linear-progress>
+
+Example:
+
+    <paper-linear-progress buffering value="40" buffer-value="70"></paper-linear-progress>
+
+Example:
+
+    <paper-linear-progress indeterminate reversed></paper-linear-progress>
+
+### Styling the linear progress element:
+
+To change the progress bar color:
+
+    paper-linear-progress {
+       --paper-linear-progress-progress-bar-color: #e91e63;
+    }
+
+To change the buffer bar color:
+
+    paper-linear-progress {
+      --paper-linear-progress-buffer-bar-color: #f8bbd0;
+    }
+
+To change the height of the element:
+
+    paper-linear-progress {
+      --paper-linear-progress-height: 2px;
+    }
+
+The following custom properties are available for styling:
+
+Custom property                               | Description                                | Default
+----------------------------------------------|--------------------------------------------|--------------
+`--paper-linear-progress-progress-bar-color`  | The color of the progress bar(s).          | `--google-blue-500`
+`--paper-linear-progress-buffer-bar-color`    | The color of the buffer bar.               | `--google-blue-100`
+`--paper-linear-progress-height`              | The height of the element.                 | `4px`
+
+> _NOTE_: The default color values for the above custom properties are provided by [paper-styles](https://github.com/PolymerElements/paper-styles).
+
+@group Paper Elements
+@element paper-progress
+@demo demo/index.html
+@hero hero.svg
+-->
+
+<dom-module id="paper-linear-progress">
+  <template>
+    <style>
+      :host {
+        --height: var(--paper-linear-progress-height, 4px);
```

Comment:
It might be a good idea to prefix this "temp" custom property with some naming convention that makes it more specific to the `paper-linear-progress` use case. The likelihood that the current name will actually cause a problem is very small, but from a conventions perspective it would be better to set a good example.